mohu - Technical Training

mizosoft opened the pull request.

vchu-odoo commented on estate/__manifest__.py:
```diff
@@ -0,0 +1,12 @@
+    'version': '0.1',
+    'summary': 'Tracks RealEstate property and buyer offers',
+    'description': "Tracks RealEstate property and buyer offers",
+    'depends': [
+        'base'
+    ],
+    'installable': True,
+    'application': True,
+    'auto_install': False
```
`installable` and `auto_install` are by default these values, no need to repeat it.
```suggestion
    'depends': ['base'],
    'application': True,
```

vchu-odoo reviewed (commented):
Good work!

Only one small nitpicking and suggestion.

vchu-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,110 @@
+
+    best_offer = fields.Float('Best Offer', compute='_compute_best_offer')
+
+    accepted_offer_id = fields.Many2one('estate.property.offer', string='Accepted Offer')
```
This field is not mentioned in the tutorial.

mizosoft replied:
answered below

vchu-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,238 @@
+        <field name="view_mode">tree,form</field>
+    </record>
+
+    <record id="estate_property_type_action" model="ir.actions.act_window">
+        <field name="name">Property Types</field>
+        <field name="res_model">estate.property.type</field>
+        <field name="view_mode">tree,form</field>
+    </record>
+
+    <record id="estate_property_tag_action" model="ir.actions.act_window">
+        <field name="name">Property Types</field>
+        <field name="res_model">estate.property.tag</field>
+        <field name="view_mode">tree,form</field>
+    </record>
```
for each model we create separate files like `estate_property_tag_view` and `estate_property_type_view` to organise code better and put there all views related to the model

mizosoft replied:
makes sense

vchu-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,238 @@
+                    <button name="action_mark_sold" type="object" string="Sold"/>
+                    <button name="action_mark_cancelled" type="object" string="Cancel"/>
+                </header>
+
```
No extra lines needed inside templates/records.
```suggestion
```

mizosoft replied:
answered below

vchu-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,238 @@
+                        </h1>
+                    </div>
+
+                    <group>
+                        <field name="status"/>
+                        <field name="property_type_id"/>
+                        <field name="postcode"/>
+                        <field name="date_availability"/>
+                        <field name="expected_price"/>  
+                        <field name="selling_price"/>
+                        <field name="best_offer"/>
+                    </group>
```
does not correspond to the description in tutorial

vchu-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,238 @@
+                <field name="facades"/>
+                <field name="property_type_id"/>
+
+                <filter string="Available" name="available" domain="['|', ('status', '=', 'new'), ('status', '=', 'offer_received')]"/>
```
Just an alternative way
```suggestion
                <filter string="Available" name="available" domain="[('status', 'in', ('new', 'offer_received'))]"/>
```

mizosoft replied:
Good suggestion!

vchu-odoo reviewed (commented):
Good job!
Continue like that :+1: 

There are some fields in estate property that do not correspond to the task description in the tutorial, is it intended?

mizosoft commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,110 @@
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        for record in self: 
+            record.best_offer = max(self.offer_ids.mapped('price')) if len(self.offer_ids) > 0 else None
```
This is nice!

mizosoft commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,110 @@
+    description = fields.Text('Description')
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date('Available From', copy=False, 
+                                    default=lambda _: date.today() + relativedelta(months=3))
```
I knew about this but I noticed that `fields.Date` is just a wrapper over `datetime.date`, so I opted for the method that was more succinct. Do you think the second way is more appropriate.

vchu-odoo replied:
We should leave lambda ofc, it is always better to use odoo framework methods for consistency, performance or security
```suggestion
                                    default=lambda _: fields.Date().add(fields.Date().today(), months=3)
```

mizosoft commented:
> There are some fields in estate property that do not correspond to the task description in the tutorial, is it intended?

Yes, I added `accepted_offer_id` to allow a NO-OP to handle accepting or refusing an already accepted offer. I've modified the code to rely on `estate_property.selling_price` & `estate_property_offer.status` instead for these cases.

vchu-odoo reviewed (commented):
Congrats on completing python tutorial, great job!
